tests/phpunit/tests/link/getEditCommentLink.php

```diff
@@ -0,0 +1,102 @@
+		);
+	}
+
+	public function set_up() {
+		parent::set_up();
+		wp_set_current_user( self::$user_ids['admin'] );
+	}
+
+	public function test_get_edit_comment_link_display_context() {
+		$comment_id   = self::$comment_ids['valid'];
+		$expected_url = admin_url( 'comment.php?action=editcomment&amp;c=' . $comment_id );
+		$actual_url   = get_edit_comment_link( $comment_id, 'display' );
+
+		$this->assertSame( $expected_url, $actual_url );
+	}
+
+	public function test_get_edit_comment_link_view_context() {
+		$comment_id   = self::$comment_ids['valid'];
+		$expected_url = admin_url( 'comment.php?action=editcomment&c=' . $comment_id );
+		$actual_url   = get_edit_comment_link( $comment_id, '' );
+
+		$this->assertSame( $expected_url, $actual_url );
+	}
+
+	public function test_get_edit_comment_link_invalid_comment() {
+		$comment_id         = self::$comment_ids['invalid'];
+		$actual_url_display = get_edit_comment_link( $comment_id, 'display' );
+		$actual_url_view    = get_edit_comment_link( $comment_id, 'view' );
+
+		$this->assertNull( $actual_url_display );
+		$this->assertNull( $actual_url_view );
+	}
+
+	public function test_get_edit_comment_link_user_cannot_edit() {
+		wp_set_current_user( self::$user_ids['subscriber'] );
+		$comment_id         = self::$comment_ids['valid'];
+		$actual_url_display = get_edit_comment_link( $comment_id, 'display' );
+		$actual_url_view    = get_edit_comment_link( $comment_id, 'view' );
+
+		$this->assertNull( $actual_url_display );
+		$this->assertNull( $actual_url_view );
+	}
+
+	/**
+	 * The test case verifies that the get_edit_comment_link function to generates comment link for editing comments,
+	 * and that the URLs are correctly filtered based on context to include HTML entities in link
+	 * $comment_id The ID of the comment to test, retrieved from self::$comment_ids['valid'].
+	 * $expected_url The expected URL format when the context is 'display', with an HTML entity for the ampersand (&amp;).
+	 * $expected_url_view The expected URL format when the context is 'view', with a regular ampersand (&).
+	 * @ticket 61727
+	 */
+
+	public function test_get_edit_comment_link_filter() {
+		$comment_id        = self::$comment_ids['valid'];
+		$expected_url      = admin_url( 'comment.php?action=editcomment&amp;c=' . $comment_id );
+		$expected_url_view = admin_url( 'comment.php?action=editcomment&c=' . $comment_id );
+
+		add_filter(
+			'get_edit_comment_link',
+			function ( $location, $comment_id, $context ) use ( $expected_url, $expected_url_view ) {
+				// Ensure the filtered URL matches the expected URL
+				if ( 'display' === $context ) {
+					$location = admin_url( 'comment.php?action=editcomment&amp;c=' ) . $comment_id;
+					$this->assertSame( $expected_url, $location );
+				} else {
+					$location = admin_url( 'comment.php?action=editcomment&c=' ) . $comment_id;
+					$this->assertSame( $expected_url_view, $location );
+				}
+					return $location; // Return unchanged
+			},
+			10,
+			3
+		);
```

Comment:
This is a bit confusing, I'm not sure what this is testing. A few notes:
* We shouldn't use assertions within filter callbacks.
* The filter callback should be used to overwrite the location with something else than the regular result. Since it's a test, you can use _any_ URL.
* You could for example use something like `admin_url( 'comment-test.php?context=' . $context )`. This way you can have two different `get_edit_comment_link()` function calls and can assert that the returned URL matches the one from the filter callback, including the passed context value.